### DISCUSSION TITLE
Simple CMS and other features made using livewire

### DISCUSSION THREAD

**Discussion by shyamsitaula** [livewire/livewire#8655] [2. Show and tell]
![dashboard-screenshot-1](https://github.com/user-attachments/assets/a35ba537-5b5a-4927-90cf-50a3f98a6635)

Looking for any feedback please.

https://github.com/oitcode/Cbswei

Any feedback is welcome.

**Comment by shyamsitaula** [comment]
Even ONE feedback would be so much appreciated. Was hoping for some feedback from livewire community. It would mean so much.

**Reply by lrljoe** [reply]
You've shared a massive project, and people won't have time to review something that large.  For future, start small, create a handful of Models, and the related CRUD elements, and people will be more likely to advise, before you create hundreds of them.

To start off - at the moment, your approach requires that someone clones your repository, and starts from there.
This will present difficulties if you release an update or similar, as changes will clash, and it prevents people from customising things smoothly.
I'd strongly suggest that you use a package approach.

You seem to be using Components and duplicating code where you shouldn't be, including validation rules, and the logic for creating/saving: for example:
BriefDescriptionCreate
BriefDescriptionEdit

You're duplicating logic, validation etc, and should be using a Livewire Form rather than a Component in of itself, and determining whether to create/update based on whether you've passed in a Model, rather than having two/three of every single component.
Livewire Forms here: [https://livewire.laravel.com/docs/forms](https://livewire.laravel.com/docs/forms)

You're not type-hinting anything, which may lead to issues down the road, for example:
```
class CustomerCreate extends Component
{
    public $name;
    public $phone;
    public $email;
    public $address;
    public $pan_num;
```
Should have types defined for $name, $phone etc.  Likewise, you should be adding your validation against those, e.g.
```
class CustomerCreate extends Component
{
    public ?string $name;
    public ?string $phone;
    public ?string $email;
    public ?string $address;
    public ?integer $pan_num;
```

You're not sanitising any inputs, which means malicious users could add Stored XSS (for example) into a field.

You've seemingly abandoned using standardised directory structures so that it's clear whether something is a Model or not purely based on location, which again will add complexities down the road.

You're not using the more up-to-date approach of Validation attributes, allowing for more straight forward live validation etc.
[https://livewire.laravel.com/docs/validation#validate-attributes](https://livewire.laravel.com/docs/validation#validate-attributes)

You're not using any kind of blade components, so you're duplicating code all over the place, 
For example, you have
```
<input type="date" class="form-control" wire:model="XYZ" />
```
In a lot of locations, which means if you wanted to update the CSS Class, you'd have to update dozens of blades rather than just one.

I can't see any evidence of Authorisation checks, i.e. gates/policies, ensuring that a user has permission to do something.

You're trying to do too much in a single "thing", segregate out the elements to minify/minimise.  Again, package approach, why would someone want CafeSale when they have nothing to do with a Cafe.

In short - I think you've got carried away before really understanding how Laravel/Livewire work, and how to make things extendable and scalable.

**Reply by shyamsitaula** [reply]
@lrljoe 

Thanks a lot for this detailed feedback. This is exactly what I was looking for. Really appreciate for the time you took to check the code and providing the feedback.

**Reply by joshhanley** [reply]
@oitcode I haven't got the time to look at your project at the moment, but I will say that all of @lrljoe's feedback is valid and are good practices to follow. 🙂

**Reply by shyamsitaula** [reply]
@joshhanley Thanks for your input too. I am working on inputs given by @lrljoe at the moment. @joshhanley I hope you get some time to check this project. Would be really glad if someone took the time out to clone this repo and actually see the admin panel in their local environment. I guess i shouldn't be asking more, and I am actually glad that I got some replies here from the community.

**Reply by lrljoe** [reply]
I think if you rationalised your code a bit, made it a bit more current with Livewire features, then people would clone and play with it.

Your inputs desperately need cleaning up, and I'd suggest you do that as a "first thing".  As well as allowing for different approaches for wire:model-ing.

Keep in mind that "defer"  is a Livewire 2x thing, and your package relies on Livewire 2.x.  Anyone starting a new project really should be using Livewire 3.x, purely from a security/support perspective.

Things like this, need to just be a blade component, and actually use the W3C conventions for indicating what the label belongs to.

```
    <div class="form-group">
      <label for="">Name</label>
      <input type="text"
          class="form-control"
          wire:model.defer="name"
          style="font-size: 1.3rem;">
      @error ('name') <span class="text-danger">{{ $message }}</span> @enderror
    </div>
```

Remove your inline style usage as well, and use classes with an externalised stylesheet, as secure CSP policies will stop those inline styles from being executed.

I think if you got everything rationalised, then people may pick it up and use it, but at the moment, it'd be way too difficult to customise/theme, plus nobody will want to start a project of this scale on Livewire 2.x

Also, review your dependencies, you should make sure that everything is on a supported and actively maintained version, for example, you've got PHP 7.x, and laravel/framework 7.x, those are both way out of date, and not something you want to try to support.

**Reply by shyamsitaula** [reply]
@lrljoe Thanks again for your feedback. Appreciate the time you are taking out to give detailed feedbacks. I am working on them. As you have noticed this project has really become very big (although boilerplate type), so it may take some time to implement them. But I have started to work on it. Thanks once again. Hopefully I will comment when I make any progress.

**Reply by shyamsitaula** [reply]
It has been upgraded to Laravel 10 and Livewire 3.

**Comment by Kslprgrm** [comment]
Nice admin panel. Also good concept of post page and team page.

**Comment by shyamsitaula** [comment]
It has been upgraded to Laravel 11 now.

**Comment by shyamsitaula** [comment]
Anyone could check it out?
